### PR TITLE
Changed to new workflow status

### LINKED ISSUE #965: Legacy job implementation in user-api that should be replaced with argo related logic [3pt]
The reason behind this is the legacy job implementation in user-api that should be replaced with argo related logic:

https://github.com/thoth-station/user-api/blob/c1b80baae49b90556534f3c5de3c5e47a0c8d530/thoth/user_api/api_v1.py#L798

It should be replaced with get_workflow_status. Mind the difference in the returned dict - they are not easily interchangeable easily.

The same applies for the other status and log endpoints.

_Originally posted by @fridex in https://github.com/thoth-station/thamos/pull/475#issuecomment-659187384_

### PR DESCRIPTION
**Pull request by saisankargochhayat**
## Related Issues and Dependencies
Closes #965 
Dependent on - https://github.com/thoth-station/common/pull/919

### CODE REVIEW

**Comment by saisankargochhayat** [comment]
@fridex How do we wanna handle when the state returned is `Succeeded`?
Also the schema `AnalysisStatusResponse`, I think the `
​/analyze​/{analysis_id}​/status` endpoint also need's to be updated to use only argo.

**Comment by fridex** [comment]
> @fridex How do we wanna handle when the state returned is `Succeeded`?

In that case raise an error, results should be available already as we ask for results first.

> Also the schema `AnalysisStatusResponse`, I think the ` ​/analyze​/{analysis_id}​/status` endpoint also need's to be updated to use only argo.

+1

**Comment by saisankargochhayat** [comment]
> +1

@fridex  I have updated the ​`/analyze​/{analysis_id}​/status` endpoint. :+1:

File: thoth/user_api/api_v1.py [hunk 1]
```diff
@@ -793,26 +775,19 @@ def _get_document(adapter_class, analysis_id: str, name_prefix: str = None, name
...
-                    # check before document retrieval - this solution is more optimal as we do not ask master
-                    # status each time.
+                status = _OPENSHIFT.get_workflow_status_report(analysis_id, namespace=namespace).get("status")
+                if status["state"] == "Running":
                     return {"error": "Analysis is still in progress", "status": status, "parameters": parameters}, 202
-                elif status["state"] == "terminated":
+                elif status["state"] == "Failed":
                     return {"error": "Analysis was not successful", "status": status, "parameters": parameters}, 400
-                elif status["state"] in ("scheduling", "waiting", "registered"):
+                elif status["state"] in ("Pending"):
```
Comment (fridex) [inline thoth/user_api/api_v1.py:783]: This will trigger string look up.

```python
>>> "o" in ("foo")
True
```

File: openapi/openapi.yaml [hunk 2]
```diff
@@ -2084,7 +2074,7 @@ components:
                 has started.
             state:
               type: string
-              example: terminated
+              example: Running
```
Comment (fridex) [inline openapi/openapi.yaml:2077]: nitpick: can we keep these states lowercase? (they can be translated in thoth-common)

File: thoth/user_api/api_v1.py [hunk 3]
```diff
@@ -344,37 +344,19 @@ def get_advise_python_log(analysis_id: str):
...
-
-    return status, code
+    if not analysis_id:
+        return {"error": "No analysis id provided.", "parameters": analysis_id}, 400
+    if not analysis_id.startswith("adviser-"):
+        return {"error": "Wrong analysis id provided", "parameters": analysis_id}, 400
+    try:
+        status = _OPENSHIFT.get_workflow_status_report(
+            analysis_id, namespace=Configuration.THOTH_BACKEND_NAMESPACE
+        ).get("status")
```
Comment (fridex) [inline thoth/user_api/api_v1.py:354]: status should not be None I guess. Also, we should return a dict here (not string) to keep API restful API.

File: openapi/openapi.yaml (hunk 2 above)
Comment (saisankargochhayat) [reply openapi/openapi.yaml:2077]: Sure, I will add a tolower() in thoth-common :+1:

File: thoth/user_api/api_v1.py (hunk 3 above)
Comment (saisankargochhayat) [reply thoth/user_api/api_v1.py:354]: status.get["status"] returns a dict, I can add a `or {}` in cases where that is none.

Comment (fridex) [reply thoth/user_api/api_v1.py:354]: I think you can directly expect "status" to be set:
```python
return OPENSHIFT.get_workflow_status_report(analysis_id, namespace=Configuration.THOTH_BACKEND_NAMESPACE)["status"], ...
```

Comment (fridex) [reply thoth/user_api/api_v1.py:354]: not nested dict would make this prettier, see https://github.com/thoth-station/common/pull/919#discussion_r461758972

File: thoth/user_api/api_v1.py (hunk 1 above)
Comment (saisankargochhayat) [reply thoth/user_api/api_v1.py]: thanks

**Comment by saisankargochhayat** [comment]
@fridex thanks, I have changed it accordingly, PTAL :smile:

**Pull request merged**